sql/core/src/main/scala/org/apache/spark/sql/ExperimentalMethods.scala

```diff
@@ -42,7 +43,7 @@ class ExperimentalMethods private[sql]() {
    * @since 1.3.0
    */
   @Experimental
-  @volatile var extraStrategies: Seq[Strategy] = Nil
+  @volatile var extraStrategies: Seq[SparkPlanner => Strategy] = Nil
```

Comment:
I'm not sure if we want to change this interface and this feels a little hacky to me.  It seems like `planLater` might actually be the problem.  In an earlier prototype we actually use a node that indicated "planLater" and looped it back through the planner instead of doing it all eagerly.  This let us do plan space exploration, but I'd have to spend some time thinking about how exactly that could work.